### DISCUSSION TITLE
Building buttons with two pieces of moveable text

### DISCUSSION THREAD

**Discussion by agillis** [HASwitchPlate/openHASP#698] [Q&A] [answered]
I want to build some buttons that look like the NSPanel project. This requires both an icon (text not an image) and the label to be specifically locale on the button (not centered).

I order to do this I needed three different objects the button, a text label, and a dummy object behind all of it the colors the button.

```
{"id": 10, "obj": "obj", "x": 5, "y": 125, "w": 150, "h": 110, "radius": 25, "border_side": 0, "bg_color": "#333"}
{"id": 11, "obj": "label", "x": 20, "y": 180, "w": 150, "h": 80, "text_color": "#B6B6B6", "text_font": 24, "text": "Livingroom"}
{"id": 12, "obj": "btn", "x": 5, "y": 125, "w": 150, "h": 110, "border_side": 0, "text_font": 24, "value_ofs_x": -45, "value_ofs_y": -17, "value_font": 32, "value_color": "#B6B6B6", "bg_opa": 0, "value_str": "\ue335"}
```

Is there a better way to do this? I can move the icon (text) but not the text on a button so I need to use a label. But if the label is in front of the button I can't press the button. So I need to make the button transparent and put the label behind it. Then I need the dummy object to color it all.

![Screenshot](https://github.com/HASwitchPlate/openHASP/assets/385810/56e53cf9-e202-4701-9395-c549c8132651)

**Comment by fvanroie** [comment, marked as answer]
Can you try setting the `click` property of the label to `0` so that it won't catch any events. Then the underlying button should still react to the presses.

**Comment by agillis** [comment]
Nice! That worked. I'm just getting started with the openHASP project and so far it is working really well!

**Comment by agillis** [comment]
This is what I ended up doing.

```
{"id": 10, "obj": "btn", "x": 5, "y": 125, "w": 150, "h": 110, "radius": 25, "border_side": 0, "bg_color": "#333", "value_ofs_x": -45, "value_ofs_y": -17, "value_font": 32, "value_color": "#B6B6B6", "value_str": "\ue335"}
{"id": 11, "obj": "label", "x": 20, "y": 180, "w": 150, "h": 80, "text_color": "#B6B6B6", "text_font": 24, "click": 0, "text": "Livingroom"}
```

**Comment by palhaca** [comment]
is it possible to use this design but with an internal gpio, ("groupid":3) where whe normally use btn?

**Reply by palhaca** [reply]
ok i found by myself the solution if you need here it is:
`{"page":0,"id": 30,"obj":"btn","x":325,"y":75,"w":150,"h":115,"toggle":true,"text":"\uE769","text_font":48,"text_color":"#e88c03","radius":25,"border_side":0,"bg_color":"#1c1c1c","click":true,"groupid":1}
{"page":0,"id": 31, "obj":"label","x":362,"y":150,"w":150,"h":40, "text_color": "#B6B6B6", "text_font": 24, "click":false, "text": "Switch 1"}`

**Comment by Marco1971Repo** [comment]
I do: 
```json
{"page":1, "parentid":81, "id":70,"obj":"btn","x":1,"y":1,"w":75,"h":75,"text":"\uE335", "text_font":16, "toggle":true}
{"page":1, "parentid":70, "id":71,"obj":"label","x":1,"y":1,"w":73,"h":18,"text":"Corridoio", "bg_opa":0, "align":"center", "text_color":"#FFFFFF", "text_font":16, "click":0}
```
Label is a child of btn and you can move btn an label around like a single thing.
